src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,159 @@
+        String[] input = line.split(" ");
+        if (input.length < DONE_MIN_LENGTH) {
+            throw new DukeException(DONE_DESCRIPTION_ERROR);
+        }
+        int index = Integer.parseInt(input[1]) - 1;
+        if (index >= currCount) {
+            throw new DukeException(DONE_EXCEED_ERROR);
+        }
+        tasks[index].markAsDone();
+        printHorizontalLine();
+        System.out.println(" Nice! I've marked this task as done:");
+        System.out.println(" " + tasks[index].toString());
+        printHorizontalLine();
+    }
+
+    public static void printList(Task[] tasks, int currCount) {
+        printHorizontalLine();
+        System.out.println(" Here are the tasks in your list:");
+        Task[] taskList = Arrays.copyOf(tasks, currCount);
+        int taskCount = 1;
+        for (Task elem : taskList) {
+            System.out.println(" " + taskCount + ". " + elem.toString());
+            taskCount += 1;
+        }
+        printHorizontalLine();
+    }
+
+    public static int completeAddTask(Task[] tasks, int currCount) {
+        printHorizontalLine();
+        System.out.println(" Got it. I've added this task:");
+        System.out.println(" " + tasks[currCount].toString());
+        int numTasks = currCount + 1;
+        System.out.println(" Now you have " + numTasks + " tasks in the list.");
+        printHorizontalLine();
+        currCount += 1;
+        return currCount;
+    }
+
+    public static void addEvent(Task[] tasks, int currCount, String line) throws DukeException {
+        if (line.length() < EVENT_MIN_LENGTH) {
+            throw new DukeException(EVENT_DESCRIPTION_ERROR);
+        }
+        if (!line.contains("/at")) {
+            throw new DukeException(EVENT_EMPTY_ERROR);
+        }
+        String[] input = line.substring(EVENT_DESCRIPTION_START).split("/");
+        tasks[currCount] = new Event(input[0], input[1].substring(EVENT_DEADLINE_TIME));
+    }
+
+    public static void addDeadline(Task[] tasks, int currCount, String line) throws DukeException {
+        if (line.length() < DEADLINE_MIN_LENGTH) {
+            throw new DukeException(DEADLINE_DESCRIPTION_ERROR);
+        }
+        if (!line.contains("/by")) {
+            throw new DukeException(DEADLINE_EMPTY_ERROR);
+        }
+        String[] input = line.substring(DEADLINE_DESCRIPTION_START).split("/");
+        // first elem: description, second elem: deadline
+        tasks[currCount] = new Deadline(input[0], input[1].substring(EVENT_DEADLINE_TIME));
+    }
+
+    public static void addTodo(Task[] tasks, int currCount, String line) throws DukeException {
+        if (line.length() < TODO_MIN_LENGTH) {
+            throw new DukeException(TODO_DESCRIPTION_ERROR);
+        }
+        // take the description part of the input string
+        tasks[currCount] = new Todo(line.substring(TODO_DESCRIPTION_START));
+    }
+
+    public static void printGoodBye() {
```

Comment:
Consider using consistent method names across similar methods. For example, since you named your welcome message method `printWelcomeMessage`, your exit message method is recommended to be `printExitMessage` so that it is more consistent and easier for other people to trace through your code. This is the same for other print message methods.